.github/CONTRIBUTING.md

```diff
@@ -88,11 +70,86 @@ export const handler = getTypedTrackHandler(async (params: Params, { events }) =
 })
 ```
 
-4) Make sure to include an example application in the [`examples/`](../examples) directory that shows how to use your client library.
+3) Implement your compiler, following the instruction under `Compiling a Typewriter Client` below.
+
+4) Most importantly, make sure to include an example program in the [`examples/`](../examples) directory that shows how to use your client library. You should use the Tracking Plan from [`examples/local-tracking-plans/tracking-plan.json`](../examples/local-tracking-plans/tracking-plan.json).
+
+5) Update the [`tests`](../tests/commands) directory to generate snapshot tests for your new language. Simply add a test file: `commands/gen-{language}/gen-{language}.test.ts` with a snapshot test:
+
+```js
+import { genGo } from '../../../src/commands/gen-go'
+import { testSnapshotSingleFile } from '../snapshots'
+
+test('genGo - compiled output matches snapshot', async () => {
+  await testSnapshotSingleFile(events => genGo(events), 'index.go')
+})
+```
+
+> If you have any questions, feel free to raise them in the issue for your language proposal. We're happy to help out!
+
+### Compiling a Typewriter Client
```

Comment:
Should this be renamed to something to indicate they'd be implementing the client generator?

"Implementing a Typewriter Client Compiler/Generator"

Reply:
Good call!